Mathlib/Data/Nat/Choose/Sum.lean

```diff
@@ -128,6 +128,24 @@ theorem sum_Icc_choose (n k : ℕ) : ∑ m ∈ Icc k n, m.choose k = (n + 1).cho
       show Ico k (n + 1) = Icc k n by rfl, ih, choose_succ_succ' (n + 1)]
   · rw [choose_eq_zero_of_lt (by omega), Icc_eq_empty_of_lt h, sum_empty]
 
+/-- Summing `(i+k).choose k` for `i ∈ [0, n]` gives `(n + (k + 1)).choose (k + 1)`.
+
+Combinatorial interpretation: `(i+k).choose k` is the number of decompositions of `[0, i)` in
+`k + 1` (possibly empty) intervals (this follows from a stars and bars description). In particular,
+`(n + (k + 1)).choose (k + 1)` corresponds to decomposing `[0, n)` into `k + 2` intervals. By
+putting away the last interval (of some length `n-i`), we have to decompose the remaining interval
+`[0, i)` into `k+1` intervals, hence the sum.
+-/
+lemma sum_range_add_choose (k n : ℕ) :
```

Comment:
Can you note that this is just a shifted version of my `sum_Icc_choose` right above?